src/background.js

```diff
@@ -36,14 +48,75 @@ function sendMetrics(event, context = {}) {
   timeouts[event] = setTimeout(later, 20000);
 }
 
+function authenticate() {
+  const fxaKeysUtil = new fxaCryptoRelier.OAuthUtils({
+    oauthServer: FXA_OAUTH_SERVER
+  });
+    chrome.runtime.sendMessage({
+      action: 'sync-opening'
+    });
+  fxaKeysUtil.launchWebExtensionKeyFlow(FXA_CLIENT_ID, {
+    redirectUri: browser.identity.getRedirectURL(),
+    scopes: FXA_SCOPES,
+  }).then((loginDetails) => {
+    const key = loginDetails.keys['https://identity.mozilla.org/apps/notes'];
+    const credentials = {
+      access_token: loginDetails.access_token,
+      refresh_token: loginDetails.refresh_token,
+      key,
+      metadata: {
+        server: FXA_OAUTH_SERVER,
+        client_id: FXA_CLIENT_ID,
+        scope: FXA_SCOPES
+      }
+    };
+    console.log('Login succeeded', credentials);
+
+    fxaFetchProfile(FXA_PROFILE_SERVER, credentials.access_token).then((profile) => {
+      browser.storage.local.set({credentials}).then(() => {
+        chrome.runtime.sendMessage({
+          action: 'sync-authenticated',
+          credentials,
+          profile
+        });
+      });
+    });
+  }, (err) => {
```

Comment:
Should this be a `.then(loginDetails, err)`, or a `then(loginDetails).catch(err)`?
**Disclaimer:** I always forget the subtle details of the `.then(a, b)` syntax, if an error in `a` triggers the error case of `b` or not.